### PR TITLE
tap-gui self-cert on windows EC2

### PR DESCRIPTION
**Pull request by satya-dillikar**
Description of Changes:

- Added googlechrome, jq, base64 packages on Windows
- Added AWS CLI and Kubectl on Windows
- Added tap-gui self-cert Root-CA Cert to Windows Cert Store
- Removed Firefox as this does not use Windows Cert Store. Google and MS Edge work perfectly with Windows Cert Store
- Increase timeout values for single-cluster to avoid wait condition timeout flakes.
- Created common QSBastionIamRole & QSBastionSecurityGroup to be used for both Ubuntu and Windows Bastion.
- Separated LinuxBastionSecurityGroup (for ssh) & WindowsBastionSecurityGroup (for RDP) have specific ports used for Ubuntu and Windows Bastion.
- Added WindowsBastion Dependency on PhaseTAPTests to start Windows EC2 only after successful TAP deployment.
- Tested TAP-GUI over HTTPS in single-cluster and multi-cluster deployments.

### CODE REVIEW

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:621]: *nit*/*non-blocking*:
I am curious which timeouts you ran into. IMHO it does not really make sense to bump the `TAPWorkloadInstall` for  singleCluster/noRelocation to 900 while leaving it at 600 for singleCluster/withRelocation. IIRC the workload installation is the same for noRelocation / withRelocation.
Also, setting the same timeouts for singleCluster/noRelocation and singelCluster/withRelocation seems a bit excessive to me, as we know that multi cluster is quite a bit slower, mostly for `PhaseCloudInit` & `PhaseTAPInstall`.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:2466]: *nit*/*non-blocking*:
```
kubectl get secrets/tap-ingress-selfsigned-root-ca -n cert-manager \
  -o go-template='{{ index  .data "tls.crt" | base64decode }} 
```
would do the same thing without external dependencies (`jq`, `base64`)

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:2474]: *nit*/*non-blocking*:
In the interest of making the stack "as fast as possible", should we maybe depend on `PhaseTAPInstall` here?
I am not sure how long it takes for the windows bastion to come up / be configured, but we could do it a bit earlier, while the workload get's deployed / the tests run.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:1077]: *nit*/*non-blocking*:
I guess we give the windows bastion host pretty much the same "permissions" now. Technically, I guess the windows bastion host would only need access to the k8s API and to the LB, and ie. not SSH. You could argue, the same is true for the linux bastion host.
Having said that, I can see that unifying the bastion hosts sec groups being easier to maintain.

**Review by hoegaarden — APPROVED** [review]
I tried to express this on multiple occasions already, and want to call it out once again:
We are introducing tight coupling of the windows bastion host and TAP here, and I think that's neither great nor necessary.
Having said that, seems I am the only one with that concern, thus won't block on that.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:621]: The present timeout value has issues, Hence I have to bump it up. I did not want to be conservative as this is just a timeout value and I happen to use the same as in multi-cluster as this is already tested and works. I know this might be a little longer for this use case but we can fine-tune it if we got more time.

**Comment by satya-dillikar** [comment]
> I tried to express this on multiple occasions already, and want to call it out once again: We are introducing tight coupling of the windows bastion host and TAP here, and I think that's neither great nor necessary. Having said that, seems I am the only one with that concern, thus won't block on that.

The present purpose of a windows instance is to access TAP-GUI and there is no point in provisioning one if the TAP-GUI is not ready.  Hence I think it is a good idea to make sure TAP is deployed properly before we can access it in Windows. If we follow the alternate solution of creating private CA and Cert, then this can be avoided. But for this use case, we are not creating a new private CA/Cert and using the existing self-cert. Hence this is required for this solution.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:2474]: I did thought of using that PhaseTAPInstall. But we do have some TAP-GUI tests after that Phase, hence I used PhaseTAPTests to be sure that all tests pass before we access TAP-GUI successfully. 
FYI, I guess Windows EC2 takes <4mins to come up & PhaseTAPTests takes just a few seconds. Hence this is not a big deal of savings and can easily change the Phase.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:2466]: thanks for the information. I shall try this.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:2474]: FWIW: I don't think the tests actually fail, ie. return a non-zero exit code and fail the deployment.
Still, I understand your point.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:1077]:
```
- Created common QSBastionIamRole & QSBastionSecurityGroup to be used for both Ubuntu and Windows Bastion.
- Separated LinuxBastionSecurityGroup (for ssh) & WindowsBastionSecurityGroup (for RDP) have specific ports used for Ubuntu and Windows Bastion.
```

As said above,

- EC2 instance needs an IAM role to access all clusters (single and multi). Hence created a common QSBastionIamRole for both EC2 instances. 
- The ingress/egress SG rules need to be the same for both EC2 to access the K8s control plane /API.  Hence create a common QSBastionSecurityGroup.
- Separated LinuxBastionSecurityGroup (for ssh) & WindowsBastionSecurityGroup (for RDP) have specific ports used for Ubuntu and Windows Bastion.  We are not adding SSH to Windows or RDP to Linux. Hope that helps to clarify.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:2474]: The tests are added as informational for now without creating any havoc. We have an opportunity to improve on embedded test cases and return fail (if required)

**Comment by satya-dillikar** [comment]
merging as second approvers/ team-members are  PTO.

**Pull request merged**
